nntrainer/tensor/tensor.cpp

```diff
@@ -633,9 +633,33 @@ Tensor &Tensor::dot(Tensor const &m, Tensor &result, bool trans,
   float *rdata = result.getData();
   const float alpha = 1.0f;
   const float beta = 0.0f;
-
   enum CBLAS_TRANSPOSE transA = trans ? CblasTrans : CblasNoTrans;
   enum CBLAS_TRANSPOSE transB = trans_m ? CblasTrans : CblasNoTrans;
+
+  /// shortcut calculation in case of vector
+  /// case1: (1 * K) X (K * 1)
+  if (M == 1 && N == 1) {
+    *rdata = sdot(K, data, 1, mdata, 1);
+    return result;
+  }
+
+  /// case2: (M * K) X (K * 1)
+  if (N == 1) {
+    sgemv(CblasRowMajor, transA, dim1, dim2, alpha, data, lda, mdata, 1, beta,
```

Comment:
similarly add that transB is ignored. If not ignored, it would generate `(M * K) X (1 * K)` which should ideally give an error. As this is just optimization and not changing functional behavior, lets add it as a documentation.

Reply:
or add a check that transB must be false (which would be the case if gemm was being used here.)

Reply:
`(1 * K) == (K * 1)` in terms of memory layout. This is considered and tested in the unittest. Thanks :)

Reply:
Oh, If you mean `(M * K) X (1 * K)` giving an error, it is handled before getting in to this line.